src/Language/PureScript/Ide/Filter.hs

```diff
@@ -77,6 +79,15 @@ declarationFilter :: (IdeDeclaration -> Text -> Bool) -> Text -> [Module] -> [Mo
 declarationFilter predicate search =
   filterModuleDecls (flip predicate search)
 
+-- | Only keeps Identifiers in the given type declarations
+declarationTypeFilter :: [D.IdeDeclaration] -> Filter
+declarationTypeFilter [] = mkFilter identity
+declarationTypeFilter decls =
+    mkFilter $ filterModuleDecls filterDecls
+    where
+      filterDecls :: IdeDeclaration -> Bool
+      filterDecls decl = D.typeDeclarationForDeclaration decl `elem` decls
```

Comment:
I think it would be nicer if we could move the filtering logic into `Language.PureScript.Ide.Filter.Declaration`. I think it could expose a function of type `[D.IdeDeclaration]-> Map ModuleName IdeDeclarationAnn -> Map ModuleName IdeDeclarationAnn` which we could then simply wrap up with mkFilter in here.

This way we can keep the Declarationfilter specific logic that we plan to add in the new added module.